index.js

```diff
@@ -22,15 +23,19 @@ function setup(fetch) {
         opts.headers.set('Host', parsed.host)
       }
       opts.redirect = 'manual'
-      parsed.hostname = await resolve(parsed.hostname)
+      parsed.host = parsed.hostname = await resolve(parsed.hostname)
+      if (parsed.port) {
+        parsed.host += `:${parsed.port}`
+      }
       url = format(parsed)
     }
     const res = await fetch(url, opts)
     if (isRedirect(res.status)) {
       const redirectOpts = Object.assign({}, opts)
       redirectOpts.headers = new Headers(opts.headers)
 
-      // per fetch spec, for POST request with 301/302 response, or any request with 303 response, use GET when following redirect
+      // Per fetch spec, for POST request with 301/302 response, or any
+      // request with 303 response, use GET when following redirect
       if (
         res.status === 303 ||
         ((res.status === 301 || res.status === 302) && opts.method === 'POST')
```

Comment:
Does this also apply to 307 and 308?

Reply:
I'm not sure, but anyways we should address that in a different PR.